CDH2.py

```diff
@@ -0,0 +1,348 @@
+import os
+from Bio import SeqIO
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+from Bio.SeqUtils import gc_fraction
+from typing import Union, Tuple, Dict
+from abc import ABC, abstractmethod
+
+
+
+def analyse_gc(records: str, min_gc: Union[int, float], max_gc: Union[int, float]) -> str:
+    """
+    Return filtered FASTQ-sequences by GC-content
+    
+    :param records: FASTQ-file
+    :type records: str
+    :param min_gc: Left boundary for GC-content filtration
+    :type min_gc: Union[int, float]
+    :tupe max_gc: Right boundary for GC-content filtration
+    :type max_gc: Union[int, float]
+    :rtype: str
+    :return: filtered FASTQ-sequences
+    """
+    return {record.id: record for record in records if (min_gc/100) <= gc_fraction(record.seq) <= (max_gc/100)}
+
+
+def filter_by_length(records: str, min_length: Union[int, float], max_length: Union[int, float]) -> str:
+    """
+    Return filtered FASTQ-sequences by GC-content
+    
+    :param records: FASTQ-file
+    :type records: str
+    :param min_gc: Left boundary for GC-content filtration
+    :type min_gc: Union[int, float]
+    :tupe max_gc: Right boundary for GC-content filtration
+    :type max_gc: Union[int, float]
+    :rtype: str
+    :return: filtered FASTQ-sequences
+    """
+
+    return {record.id: record for record in records if min_length <= len(record.seq) <= max_length}
+
+
+def filter_by_quality(records: str, quality_threshold: Union[int, float]) -> str:
+    """
+    Return filtered FASTQ-sequences by quality
+    :param records: Sequnces filtered by GC-content and length
+    :type records: str
+    :param quality_threshold: boundary for quality filtration
+    :type quality_threshold: Union[int, float]
+    :rtype: str
+    :return: filtered FASTQ-sequences   
+    """
+    
+    return {record.id: record for record in records if min(record.letter_annotations["phred_quality"]) >= quality_threshold}
+
+
+
+def write_filtered_sequences_to_fastq(filtered_sequences: Dict[str, str], unfiltered_sequences: Dict[str, str],
+                                      output_file: str, folder_path: str = 'fastq_filtrator_results') -> str:
```

Comment:
В докстриге все-таки returning type указан вернее :)
```suggestion
def write_filtered_sequences_to_fastq(filtered_sequences: Dict[str, SeqRecord], unfiltered_sequences: Dict[str, SeqRecord], output_file: str, folder_path: str = 'fastq_filtrator_results') -> None:
```